lib/ansible/modules/cloud/amazon/iam_role_facts.py

```diff
@@ -0,0 +1,201 @@
+          sample: arn:aws:iam::123456789012:policy/AnsibleTestEC2Policy
+        policy_name:
+          description: Name of managed policy
+          returned: always
+          type: string
+          sample: AnsibleTestEC2Policy
+    path:
+      description: Path of role
+      returned: always
+      type: string
+      sample: /
+    role_id:
+      description: Amazon Identifier for the role
+      returned: always
+      type: string
+      sample: AROAII7ABCD123456EFGH
+    role_name:
+      description: Name of the role
+      returned: always
+      type: string
+      sample: AnsibleTestRole
+'''
+
+try:
+    import botocore
+except ImportError:
+    pass  # caught by AnsibleAWSModule
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import boto3_conn, get_aws_connection_info, ec2_argument_spec, AWSRetry
+from ansible.module_utils.ec2 import camel_dict_to_snake_dict
+
+
+@AWSRetry.exponential_backoff()
+def list_iam_roles_with_backoff(client, **kwargs):
+    paginator = client.get_paginator('list_roles')
+    return paginator.paginate(**kwargs).build_full_result()
+
+
+@AWSRetry.exponential_backoff()
+def list_iam_role_policies_with_backoff(client, role_name):
+    paginator = client.get_paginator('list_role_policies')
+    return paginator.paginate(RoleName=role_name).build_full_result()['PolicyNames']
+
+
+@AWSRetry.exponential_backoff()
+def list_iam_attached_role_policies_with_backoff(client, role_name):
+    paginator = client.get_paginator('list_attached_role_policies')
+    return paginator.paginate(RoleName=role_name).build_full_result()['AttachedPolicies']
+
+
+def describe_iam_role(module, client, role):
+    name = role['RoleName']
+    try:
+        role['InlinePolicies'] = list_iam_role_policies_with_backoff(client, name)
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        module.fail_json_aws(e, msg="Couldn't get inline policies for role %s" % name)
+    try:
+        role['ManagedPolicies'] = list_iam_attached_role_policies_with_backoff(client, name)
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        module.fail_json_aws(e, msg="Couldn't get managed  policies for role %s" % name)
+    return role
+
+
+def describe_iam_roles(module, client):
+    name = module.params['name']
+    path_prefix = module.params['path_prefix']
+    if name:
+        try:
+            roles = [client.get_role(RoleName=name)['Role']]
+        except botocore.exceptions.ClientError as e:
+            if e.response['Error']['Code'] == 'NoSuchEntity':
+                return []
+            else:
+                module.fail_json_aws(e, msg="Couldn't get IAM role %s" % name)
+        except botocore.exceptions.BotoCoreError as e:
+            module.fail_json_aws(e, msg="Couldn't get IAM role %s" % name)
+    else:
+        params = dict()
+        if path_prefix:
+            params['PathPrefix'] = path_prefix
```

Comment:
http://docs.aws.amazon.com/IAM/latest/APIReference/API_ListRoles.html

I think we should enforce this so we can give a nice error for people that don't submit proper paths, or auto-fill it so the path is valid. Paths like `foobar/` and `/foobar` will break, the path *must* start and end with `/`. Up to you, but I'd just prepend and append `/` if there isn't one there already.

Reply:
Ping @willthames - other than this validation fix this is good to go.

Reply:
@ryansb - thanks for that (and the ping) - I've updated as suggested.